src/local_scheduler.rs

```diff
@@ -288,8 +289,11 @@ impl NativeScheduler for LocalScheduler {
         let my_attempt_id = self.attempt_id.fetch_add(1, Ordering::SeqCst);
         let event_queues = self.event_queues.clone();
         let task = bincode::serialize(&task).unwrap();
-        thread_pool.execute(move || {
-            LocalScheduler::run_task::<T, U, F>(event_queues, task, id_in_job, my_attempt_id)
+
+        // run in async executor
+        let executor = &mut *env::Env::get().async_rt.lock();
```

Comment:
this locks the runtime therefore no tasks are run parallely

Reply:
I can't say I am 100% sure cause I am no expert on Tokio but that's not how it works I believe.This is essentially like `#[tokio::main]` and the only and single entry point for the runtime context and it requires a &mut ref (so it's always handed in a single place and essentially 'locked' it doesn't matter if it's behind a Mutex or not). Beware `block_on` is never again called in parallel, because it can be only called once per runtime (is not possible) at the same time until the 'main'  future in run into completion. Any future awaited within the context will be then schedulled to the internal tokio threadpool then, it doesn't require a direct reference to the runtime.

Any parallelism that we want to be achieved has to be handled through the tokio tasks themselved via `spawn`and doing a `join_all` afterwards (for example), there is an example of how it's done in ShuffleFetcher. This would be analogous to something like:
 ```
vec![task; N].into_iter().for_each(|task| spawn::thread(move || { task.run() ...)
```
but without the spawn part as the threadpool is already running and the tasks will be schedulled to them.

Reply:
Or put in a shorter way... Runtime is just a facade for the real Context which Tokio handles internally to schedule and run tasks in parallel. This is why you can always get handles (& refs) to the context which you can use to spawn futures internally even though your main entry point is locked throught a `&mut` if it makes sense.

Reply:
Ok, never mind, I think I udnerstand what you mean... actually it should be called there, we should enter the async context up the stack on `run_job` not on `submit_Task`. I had to change this becaus eit was a quick band-aid but somehow forgot about it!